storage/isolation_manager.rs

```diff
@@ -243,21 +237,16 @@ impl IsolationManager {
     }
 
     pub(crate) fn watermark(&self) -> SequenceNumber {
-        let (window, slot_index) = self.timeline.try_get_window(self.timeline.watermark()).unwrap();
-        match window.get_status(slot_index) {
-            CommitStatus::Applied(seq, _) | CommitStatus::Aborted(seq) => seq.clone(),
-            CommitStatus::Validated(_, _) | CommitStatus::Pending(_, _) | CommitStatus::Empty =>
-                unreachable!("The isolation manager must always have the watermark commit record in the window, and it must be either committed or aborted."),
-        }
+        self.timeline.watermark()
```

Comment:
Since there's no longer a mapping to be done between sequence number & relative index, we don't require an entry in the timeline at the watermark. Instead, we just return the value stored in the watermark.